### PR TITLE
Setup tests in presence tracker

### PR DESCRIPTION
**Pull request by sonalideshpandemsft**
[AB#1055](https://dev.azure.com/fluidframework/internal/_sprints/taskboard/Team%20Sasha/internal/Team%20Sasha%20-%202023.06-A?workitem=1055)

### CODE REVIEW

File: experimental/framework/data-objects/src/signaler/test/signaler.spec.ts [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+		const receivedPayload = signaler.onSignal(signalName, () => {
+			listener(payload);
+		});
+
+		assert(
+			receivedPayload !== undefined || receivedPayload !== null,
+			"Payload is not empty/undefined",
+		);
+	});
+
```
Comment (ChumpChief) [inline experimental/framework/data-objects/src/signaler/test/signaler.spec.ts:92]: Consider also adding a test that checks the incoming signal's "local" flag and clientId is correct when two clients are sending/receiving signals.

File: examples/apps/presence-tracker/tests/index.ts [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+const containerSchema: ContainerSchema = {
+	initialObjects: {
+		/* [id]: DataObject */
+		signaler: Signaler,
+	},
+};
+
+async function setup() {
+	// Get or create the document depending if we are running through the create new flow
+	const client = new TinyliciousClient();
```
[the remark this answers is not in the document]
Comment (sonalideshpandemsft) [reply examples/apps/presence-tracker/tests/index.ts:26]: I was running tinylicious locally and forgot to update the script. I'll switch these tests to use local-driver

File: examples/apps/presence-tracker/src/FocusTracker.ts [hunk 3]
```diff
@@ -3,22 +3,34 @@
...
+/**
+ * ITinyliciousUser describes the default format for users created through the Tinylicious server.
+ * However, the specific details for a user may differ depending on the server.
+ */
+export interface ITinyliciousUser {
+	id: string;
+	name: string;
+}
+
+export class FocusTracker extends DataObject implements IFocusTracker {
```
Comment (ChumpChief) [inline examples/apps/presence-tracker/src/FocusTracker.ts:33]: Choosing to make FocusTracker a DataObject means it becomes part of the persisted state of the container - I'm not sure if this is something we want?  At minimum it is a pretty significant change in the purpose of the demo.

**Review by ChumpChief — COMMENTED** [review]
I'm a bit surprised to see the change to make these DataObjects -- I think the tests look fine, but not sure we want to require persisting data for these scenarios?

**Comment by sonalideshpandemsft** [comment]
closing in favor of https://github.com/microsoft/FluidFramework/pull/16274

**Pull request closed without merging**